lucille-core/src/main/java/com/kmwllc/lucille/util/CamelCaseConfigConverter.java

```diff
@@ -0,0 +1,114 @@
+    StringBuilder sb = new StringBuilder();
+
+    try (BufferedReader br = new BufferedReader(new FileReader(file))) {
+      String line;
+      // iterating over lines for the edge case where the stage property is used one or more times as a property value in the same line
+      // in this case we want to replace only the first occurrence -- which is the stage property
+      // StringUtils.replaceEach does not give a way to limiting the number of replacements of the same value in the same line/config
+      while ((line = br.readLine()) != null) {
+        for (String stageProperty : stageProperties) {
+          if (line.contains(stageProperty)) {
+            line = StringUtils.replaceOnce(line, stageProperty, snakeToCamel(stageProperty));
+          }
+        }
+        sb.append(line).append("\n");
+      }
+    } catch (Exception e) {
+      throw new IOException("Error processing file: " + file.getName(), e);
+    }
+
+    return sb.toString();
+  }
+
+  private static void processFile(String filePath) throws Exception {
+    String newFileName = FilenameUtils.getBaseName(filePath) + "CamelCase" + ".conf";
+    String newFilePath = Paths.get(FilenameUtils.getFullPath(filePath), newFileName).toString();
+    File file = new File(filePath);
+    // collect all stageProperty that needs to change
+    Config config = ConfigFactory.parseFile(file);
+    Map<String, Object> configAsMap = config.root().unwrapped();
+    List<String> stageProperties = getStagePropertiesFromConfig(configAsMap);
+
+    // update stage properties in new file and write to newFilePath
+    String configStr = applyCamelCase(file, stageProperties);
+    writeToConfFile(configStr, newFilePath);
+  }
+
+  private static List<String> getStagePropertiesFromConfig(Map<String, Object> configAsMap) throws IOException {
+    List<String> stageProperties = new ArrayList<>();
+    try {
+      List<Map<String, Object>> pipelinesMap = (List<Map<String, Object>>) configAsMap.get("pipelines");
+      for (Map<String, Object> pipeline : pipelinesMap) {
+        List<Map<String, Object>> stages = (List<Map<String, Object>>) pipeline.get("stages");
+        for (Map<String, Object> stage : stages) {
+          List<String> properties = stage.keySet().stream()
+              // to collect only those properties that only need to be updated
+              .filter(key -> key.contains("_"))
+              .toList();
+          stageProperties.addAll(properties);
+        }
+      }
+    } catch (Exception e) {
+      throw new IOException("Error getting stage properties from config.", e);
+    }
+
+    return stageProperties;
+  }
+
+  private static void validateArguments(String[] args) {
+    if (args.length != 1) {
+      System.exit(1);
+    }
+
+    String filePath = args[0];
+    if (!filePath.endsWith(".conf")) {
+      log.error("Invalid file extension, must be used on a config file.");
+      System.exit(1);
+    }
+  }
+
+  public static void main(String[] args) {
```

Comment:
To run this I had to do mvn clean install at the top level, then go to lucille-core/target and run `java -cp 'lucille.jar:lib/*' com.kmwllc.lucille.util.CamelCaseConfigConverter`
Should document that in the class-level javadoc (or a separate readme.md file)

with no args, the utility did nothing (would be better to print out a warning message)
with a file that didn't exist, the error messages weren't easy to understand -- two cases to consider, the filename has no extension, or it does. If the filename has no extension but also doesn't exist, would be better to warn that it doesn't exist...

$ java -cp 'lucille.jar:lib/*' com.kmwllc.lucille.util.CamelCaseConfigConverter

$ java -cp 'lucille.jar:lib/*' com.kmwllc.lucille.util.CamelCaseConfigConverter 1
24/11/05 12:11:04 ERROR CamelCaseConfigConverter: Invalid file extension, must be used on a config file.

$ java -cp 'lucille.jar:lib/*' com.kmwllc.lucille.util.CamelCaseConfigConverter 1.conf
24/11/05 12:24:20 ERROR CamelCaseConfigConverter: Error processing file: 
java.io.IOException: Error getting stage properties from config.
	at com.kmwllc.lucille.util.CamelCaseConfigConverter.getStagePropertiesFromConfig(CamelCaseConfigConverter.java:86) ~[lucille.jar:0.4.1-SNAPSHOT]
	at com.kmwllc.lucille.util.CamelCaseConfigConverter.processFile(CamelCaseConfigConverter.java:64) ~[lucille.jar:0.4.1-SNAPSHOT]
	at com.kmwllc.lucille.util.CamelCaseConfigConverter.main(CamelCaseConfigConverter.java:108) [lucille.jar:0.4.1-SNAPSHOT]
Caused by: java.lang.NullPointerException: Cannot invoke "java.util.List.iterator()" because "pipelinesMap" is null
	at com.kmwllc.lucille.util.CamelCaseConfigConverter.getStagePropertiesFromConfig(CamelCaseConfigConverter.java:75) ~[lucille.jar:0.4.1-SNAPSHOT]
	... 2 more

Reply:
updated validateArguments to log error if no arguments were provided and also now checks if file exists before checking file extension in [63261d5](https://github.com/kmwtechnology/lucille/pull/203/commits/63261d5ae1fc5ee23031a4e9f3c14d620c7e13d8)